Core/GDCore/IDE/ProjectExposer.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * GDevelop Core
+ * Copyright 2008-2016 Florian Rival (Florian.Rival@gmail.com). All rights
+ * reserved. This project is released under the MIT License.
+ */
+#ifndef GDCORE_PROJECTEXPOSER_H
+#define GDCORE_PROJECTEXPOSER_H
+namespace gd {
+class Project;
+class String;
+class EventsFunctionsExtension;
+class EventsFunction;
+class EventsBasedBehavior;
+class EventsBasedObject;
+class ArbitraryEventsWorker;
+class ArbitraryEventsWorkerWithContext;
+class ArbitraryFunctionsWorker;
+class ArbitraryObjectsWorker;
+class ArbitraryEventBasedBehaviorsWorker;
+class ArbitrarySharedDataWorker;
+} // namespace gd
+
+namespace gd {
+
+/**
+ * \brief Expose a subset of the project to workers.
+ */
+class GD_CORE_API ProjectExposer {
```

Comment:
Sorry it's a bit of dumb renaming work, but could we name this:

```suggestion
class GD_CORE_API ProjectWalker {
```

or ProjectBrowser?

Because for a native english speaker, "Exposer" will sound.. super weird. 
This is a class that walk/traverse/browse the project and "expose" the elements to the worker (a.k.a visitor in the visitor pattern).